tests/json_validation_test.py

```diff
@@ -8,57 +8,42 @@
 
 """
 
-
+from utils import DATA_DIR
 from jsonschema import validate, ValidationError
+
+import json
 import glob
 import os
-import json
-from gdcdictionary import ROOT_DIR
-
-from utils import validate_entity, BaseTest
-
-DATA_DIR = os.path.join(ROOT_DIR, 'examples')
-
-class JsonValidationTests(BaseTest):
-
-    def test_valid_files(self):
-        for path in glob.glob(os.path.join(DATA_DIR, 'valid', '*.json')):
-            print "Validating {}".format(path)
-            doc = json.load(open(path, 'r'))
-            print(doc)
-            if type(doc) == dict:
-                self.add_system_props(doc)
-                validate_entity(doc, self.dictionary.schema)
-            elif type(doc) == list:
+import pytest
+
+
+def get_all_docs(subdir):
+    for path in sorted(glob.glob(os.path.join(DATA_DIR, subdir, '*.json'))):
+        with open(path, 'r') as f:
+            doc = json.load(f)
+            if isinstance(doc, dict):
+                yield doc
+            elif isinstance(doc, list):
+                if len(doc) != 1:
+                    assert False, "Mismatch between files and documents in {}".format(f.name)
                 for entity in doc:
-                    self.add_system_props(entity)
-                    validate_entity(entity, self.dictionary.schema)
+                    yield entity
             else:
-                raise Exception("Invalid json")
+                assert False, "Error collecting file: {}".format(f)
 
-    def test_invalid_files(self):
-        for path in glob.glob(os.path.join(DATA_DIR, 'invalid', '*.json')):
-            print "Validating {}".format(path)
-            doc = json.load(open(path, 'r'))
-            if type(doc) == dict:
-                self.add_system_props(doc)
-                with self.assertRaises(ValidationError):
-                    validate_entity(doc, self.dictionary.schema)
-            elif type(doc) == list:
-                for entity in doc:
-                    self.add_system_props(entity)
-                    with self.assertRaises(ValidationError):
-                        validate_entity(entity, self.dictionary.schema)
-            else:
-                raise Exception("Invalid json")
+def get_all_files(subdir):
+    file_names = []
+    for path in sorted(glob.glob(os.path.join(DATA_DIR, subdir, '*.json'))):
+        with open(path, 'r') as f:
+            file_names.append(f.name)
+    return file_names
+
+
+@pytest.mark.parametrize("doc", get_all_docs("valid"), ids=get_all_files("valid"))
+def test_valid_examples(doc, schema):
+    validate(doc, schema[doc["type"]])
```

Comment:
The `ids` seems to be a list of file names, not sure how that plays with `jsonschema.validate`

Reply:
That's so pytest doesn't say the tests are being parametrized off the word docX (where X is a number)

It is not used within the test function.

Reply:
Its used by pytest for formatting and reporting